[CI] tasks.list/10_basic/tasks_list test failure

Reproducible in `main`. Muted by https://github.com/opensearch-project/OpenSearch/commit/fbb2725ec1cde01bac8c9d5eb70d844f918b4d6b

```
./gradlew ':qa:mixed-cluster:v2.1.0#mixedClusterTest' --tests "org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=tasks.list/10_basic/tasks_list test}" -Dtests.seed=B4F39ACD8A2A3DED -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-PE -Dtests.timezone=Australia/ACT -Druntime.java=17
```

```
org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=tasks.list/10_basic/tasks_list test} FAILED
    java.lang.AssertionError: Failure at [tasks.list/10_basic:15]: field [nodes.$node_id.roles] doesn't have a true value
    Expected: not null
         but: was null
```

See the Gradle check log for detailed error message:
[Log 4153](<https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4153.log>)
[Reports 4153](<https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4153_reports.zip>)
(Comes from https://github.com/opensearch-project/OpenSearch/pull/2660#issuecomment-1088064862)
[Log 4160](<https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4160.log>)
[Reports 4160](<https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4160_reports.zip>)
(Comes from https://github.com/opensearch-project/OpenSearch/pull/2752#issuecomment-1088125675)

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/tasks.list/10_basic.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/tasks.list/10_basic.yml
@@ -1,8 +1,6 @@
 ---
 "tasks_list test":
   - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/opensearch-project/OpenSearch/issues/2757"
       features: [arbitrary_key]
 
   - do:

--- a/server/src/main/java/org/opensearch/tasks/TaskInfo.java
+++ b/server/src/main/java/org/opensearch/tasks/TaskInfo.java
@@ -142,7 +142,7 @@ public final class TaskInfo implements Writeable, ToXContentFragment {
         }
         parentTaskId = TaskId.readFromStream(in);
         headers = in.readMap(StreamInput::readString, StreamInput::readString);
-        if (in.getVersion().onOrAfter(Version.V_2_0_0)) {
+        if (in.getVersion().onOrAfter(Version.V_2_1_0)) {
             resourceStats = in.readOptionalWriteable(TaskResourceStats::new);
         } else {
             resourceStats = null;
@@ -164,7 +164,7 @@ public final class TaskInfo implements Writeable, ToXContentFragment {
         }
         parentTaskId.writeTo(out);
         out.writeMap(headers, StreamOutput::writeString, StreamOutput::writeString);
-        if (out.getVersion().onOrAfter(Version.V_2_0_0)) {
+        if (out.getVersion().onOrAfter(Version.V_2_1_0)) {
             out.writeOptionalWriteable(resourceStats);
         }
     }

